### ISSUE TITLE
about the experiments on natural image

### ISSUE THREAD

**Issue by Lupin123123** [mandt-lab/empirical-RD-sandwich#2]
If I want to evaluate the RD function of a given image dataset, why the code should train on a different dataset, like coco?

**Comment by yiboyang** [comment]
Hi, 

Good question.  I'm assuming you're training a latent variable model to estimate an upper bound on R(D). The answer basically depends on the size of your dataset and has to do with overfitting/generalization, which is core to machine learning. 

If the dataset is very large (e.g., if you train on small random crops created on the fly) so there's no risk of overfitting, then it should be OK to train on that, and it's still a good idea to estimate the R(D) on a separate test set. However, if the dataset is too small, there is a risk that your model will overfit and not give you a useful estimate of the R-D function. A finite dataset of images is technically a discrete distribution, and its R(D) (for a finite distortion function $\rho$ that satisfies $\rho(x,x)=0$) will always lie below the data entropy, which is `log(n)` bits/sample if `n` is the size of the dataset. For example, `n=24` for Kodak, so it will take no more than `R(0) = log(24) = 4.6` bits at the minimum to transmit a Kodak image *perfectly* (`D=0`), which can be achieved by storing the 24 images on the decoder side; the minimum bit-rate is even less if we allow lossy reconstruction (`D>0`). 
However, this is typically not what we want. We are instead interested in the "true" and "infinite" image-generating distribution that gave rise to the finite dataset that we observed. Therefore it's a good idea to train the R-D estimator model on a large dataset that is 

1. representative of the test data you have in mind; and 
2. has enough samples so that the model will not overfit (this can be checked by using a held-out set). 

Coco meets these criteria and is one of the most popular datasets for training neural image compressors, so it was used. But if the goal is to really estimate the compressibility of images that are similar in style to Kodak, then it'd make sense to train on some other large collection of images similar to Kodak. 

Note that *any* "beta-VAE" model $(Q_Z, Q_{X|Z}, \omega)$ can be used to estimate a point on the R(D) upper bound of  your target dataset, by evaluating the rate and distortion terms of the R-D Lagrangian on your data. This holds regardless of what kind of data the model has been trained on (the model can even be untrained; see Theorem A.3 and Corollary A.3.1 in the [paper](https://arxiv.org/pdf/2111.12166)); however the resulting R(D) bound generally will not be tight or useful.

**Comment by Lupin123123** [comment]
Hi,

Thank you for your response—your comments have been incredibly insightful.

I still have some questions regarding the estimation of the RD function. Specifically, how many samples are sufficient to reliably estimate it? I conducted experiments on the a customed dataset comprised of 8,000 samples with ResNet-VAE, but I’m concerned about the potential risk of overfitting. Additionally, if the model complexity increases—for instance, by adding more layers and channels to ResNet-VAE—would a larger number of samples be required?

Recently, I have been experimenting with ResNet-VAE to compute the upper bound of an image source. I observed that increasing the network complexity (e.g., adding more layers and channels) leads to an improvement in the upper bound curve (PSNR vs. BPP). However, I’m worried that this improvement might indicate overfitting due to the higher complexity of the model. Could you provide insights on these concerns?

**Comment by Lupin123123** [comment]
Hi,
I was also wondering whether the estimated rate-distortion function (or its upper bound) obtained by a VAE is achievable for an arbitrary source?
